### ISSUE TITLE
Numerous errors when trying to change the crosshair ingame

### ISSUE THREAD

**Issue by garfielfing** [mcolour/colourhud#11]
Here's a portion of the error log but it seems like every error is the same, repeated numerous times. This happens inconsistently. It doesn't seem to happen in the main menu or itemtest, but in other servers it typically does. This causes the crosshair to disappear and its settings to be reset.

KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*188*), (*-16*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*188*), (*-16*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*188*), (*-16*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*188*), (*-16*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*188*), (*-16*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*188*), (*-16*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*188*), (*-16*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*188*), (*-16*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*188*), (*-16*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*188*), (*-16*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*188*), (*-16*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*HudChat*), (*1*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*HudChat*), (*1*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*HudChat*), (*1*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*HudChat*), (*1*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*HudChat*), (*1*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*HudChat*), (*1*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*HudChat*), (*1*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*HudChat*), (*1*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*HudChat*), (*1*), (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: RecursiveLoadFromBuffer:  got } in key in file resource/UI/../../../../cfg/colourhud_settings.txt
WeaponNameLabel, ypos, (*if_mvm*), (*ypos*), (*PlayerName*), 
KeyValues Error: RecursiveLoadFromBuffer:  got } in key in file resource/UI/../../../../cfg/colourhud_settings.txt
0, 190, FreezeLabelKiller, (*PlayerName*), 
KeyValues Error: RecursiveLoadFromBuffer:  got } in key in file resource/UI/../../../../cfg/colourhud_settings.txt
0, {, (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*0*), (*{*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*0*), (*{*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*0*), (*{*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*AvatarImage*), (*tall*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*AvatarImage*), (*tall*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*AvatarImage*), (*tall*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*AvatarImage*), (*tall*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*Player2Name*), (*ypos*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: RecursiveLoadFromBuffer:  got } in key in file resource/UI/../../../../cfg/colourhud_settings.txt
{, ypos, (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*{*), (*ypos*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*{*), (*ypos*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*{*), (*ypos*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*{*), (*ypos*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*Player3Name*), (*ypos*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*Player3Name*), (*ypos*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*Player3Name*), (*ypos*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*Player3Name*), (*ypos*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*Player3Name*), (*ypos*), (*190*), (*FreezeLabelKiller*), (*PlayerName*), 
KeyValues Error: LoadFromBuffer: missing { in file resource/UI/../../../../cfg/colourhud_settings.txt
(*Spacer*), (*wide_minmode*), (*issue*), (*tall*), (*playernameshadow*), (*visible*), (*RedPlayerListParent*), (*RedPlayerList*),

**Comment by mcolour** [comment]
After doing testing of the hud on 3 separate machines
1) Windows PC with non-clean install of TF2 (including mastercomfig)
2) Windows PC with clean install (including mastercomfig)
3) Linux PC with clean install (including mastercomfig)

I've found out that this error of writing hud's settings incorrectly (TF2 writing variables and curly brackets out of intended order) occurs only on Linux. I've managed to negate the error, but it still occurs if the settings are changed too quickly. Try to use this version of the hud:
[colourhud-master.zip](https://github.com/mcolour/colourhud/files/13198864/colourhud-master.zip)

**Comment by garfielfing** [comment]
I might've been changing them too quickly. It also didn't seem to happen on itemtest, just Valve servers.

**Comment by mcolour** [comment]
Yeah, I've tested the hud on Valve server and on community servers as well. In the main menu everything works as intended regardless of how frequently the settings are changed. On Valve and on community servers this error starts occuring on Linux version of the game if the settings are changed frequently. However with the applied solution that I've provided in this issue this error doesn't occur, if the settings are changed approximately once per 10 seconds.
